### ISSUE TITLE
`@Immutable @EncodeNulls` doesn’t work as expected

### ISSUE THREAD

**Issue by Brett-Best** [GottaGetSwifty/CodableWrappers#24]
Hey @GottaGetSwifty,

Using `@Immutable` with `@EncodeNulls` doesn’t work as expected.

```swift
struct TestModel: Encodable {
    @EncodeNulls var date1: Date?
    @Immutable @EncodeNulls var date2: Date?
}

 print(String(data: try! JSONEncoder().encode(TestModel(date1: nil, date2: nil)), encoding: .utf8))

// OUTPUT: "{\"date1\":null}"
```

I also can’t seem to get this snippet to compile (related?):
```swift
@Immutable @OptionalCoding<DateFormatterCoding<RFC3339DateCoder>> @EncodeNulls var myDate: Date?
```

**Comment by jayrhynas** [comment]
Hopefully fixed by #25

**Comment by GottaGetSwifty** [comment]
@Brett-Best @jayrhynas Included in 2.0.7. Please Let me know if your issue is not fixed.

**Comment by Brett-Best** [comment]
Thanks so much for having a look at this @jayrhynas! Much appreciated!!

Can confirms the PR fixes my use case :D
